src/lib/services/openapi-service.ts

```diff
@@ -0,0 +1,66 @@
+import openapi from '@wesleytodd/openapi';
+import { Express, RequestHandler } from 'express';
+import { OpenAPIV3 } from 'openapi-types';
+import { IUnleashConfig } from '../types/option';
+import { createOpenApiSchema } from '../openapi';
+import {
+    AdminApiOperation,
+    ClientApiOperation,
+    ExpressOpenApi,
+} from '../openapi/types';
+
+export class OpenApiService {
+    private readonly config: IUnleashConfig;
+
+    private readonly api: ExpressOpenApi;
+
+    constructor(config: IUnleashConfig) {
+        this.config = config;
+        this.api = openapi(
+            this.docsPath(),
+            createOpenApiSchema(config.server?.unleashUrl),
+        );
+    }
+
+    // Create request validation middleware for an admin or client path.
+    validPath(op: AdminApiOperation | ClientApiOperation): RequestHandler {
+        return this.api.validPath(op);
+    }
+
+    // Serve the OpenAPI JSON at `${baseUriPath}/docs/openapi.json`,
+    // and the OpenAPI SwaggerUI at `${baseUriPath}/docs/openapi`.
+    useDocs(app: Express): void {
+        app.use(this.api);
+        app.use(this.docsPath(), this.api.swaggerui);
+    }
+
+    // The OpenAPI docs live at `<baseUriPath>/docs/openapi{,.json}`.
+    docsPath(): string {
+        const { baseUriPath = '' } = this.config.server ?? {};
+        return `${baseUriPath}/docs/openapi`;
+    }
+
+    // Add custom schemas to the generated OpenAPI spec.
+    // Used by unleash-enterprise to add its own schemas.
+    registerCustomSchemas(schemas: {
```

Comment:
Nice. I was going to ask how we'd extend Enterprise as well to make sure we had openapi for the enterprise endpoints as well :)